synfig-core/src/synfig/module.cpp

```diff
@@ -80,51 +71,46 @@ Module::subsys_init(const String &prefix)
 
 	// user's synfig library path
 #ifdef _WIN32
-	if(const char *localappdata = getenv("%LOCALAPPDATA%")) {
-		std::string user_module_path = Glib::locale_from_utf8(localappdata) + "/synfig/modules";
-		add_search_dir(user_module_path.c_str());
+	std::string localappdata = Glib::getenv("%LOCALAPPDATA%");
+	if (!localappdata.empty()) {
+		std::string user_module_path = localappdata + "/synfig/modules";
+		add_search_dir(user_module_path);
 	}
+#elifdef __APPLE__
+	std::string home = Glib::getenv("HOME");
+	if (!home.empty())
+		add_search_dir(home + "/Library/Application Support/org.synfig.SynfigStudio/modules");
+
+	add_search_dir("/Library/Frameworks/synfig.framework/Resources/modules");
 #else
-#ifdef __APPLE__
-	if(const char *home = getenv("HOME"))
-		add_search_dir(strprintf("%s/Library/Application Support/org.synfig.SynfigStudio/modules", home).c_str());
-#else
-	if(const char *home = getenv("HOME"))
-		add_search_dir(strprintf("%s/.local/share/synfig/modules", home).c_str());
-#endif
+	std::string home = Glib::getenv("HOME");
+	if (!home.empty())
+		add_search_dir(home + "/.local/share/synfig/modules");
 #endif
 
 	// (runtime) prefix path
-	add_search_dir((Glib::locale_from_utf8(prefix) + "/lib/synfig/modules").c_str());
+	add_search_dir((Glib::locale_from_utf8(prefix) + "/lib/synfig/modules"));
 
 	// path defined on build time
 #ifdef LIBDIR
 	add_search_dir(LIBDIR"/synfig/modules");
 #endif
-#ifdef __APPLE__
-	add_search_dir("/Library/Frameworks/synfig.framework/Resources/modules");
-#endif
 
 	// current working path...
 	add_search_dir(".");
-#endif
-	book_=new Book;
+
 	return true;
 }
 
 bool
-Module::subsys_stop()
+synfig::Module::subsys_stop()
 {
-	delete book_;
```

Comment:
`book_.clear()`?

Reply:
Yeah, I thought about it. But `synfig::Module` is a singleton-like object, and `book_` is a static variable, so I'm not sure it's necessary.